src/main/java/org/spongepowered/api/world/extent/BlockVolume.java

```diff
@@ -51,4 +60,231 @@
+     * @return A light level, nominally between 0 and 15, inclusive
+     */
+    byte getLuminance(Vector3i position);
+
+    /**
+     * Get the light level for this object that is caused by an overhead sky.
+     *
+     * <p>Higher levels indicate a higher luminance. If no sky is overheard,
+     * the return value may be 0.</p>
+     *
+     * @param position The position of the block
+     * @return A light level, nominally between 0 and 15, inclusive
+     */
+    byte getLuminanceFromSky(Vector3i position);
+
+    /**
+     * Get the light level for this object that is caused by everything
+     * other than the sky.
+     *
+     * <p>Higher levels indicate a higher luminance.</p>
+     *
+     * @param position The position of the block
+     * @return A light level, nominally between 0 and 15, inclusive
+     */
+    byte getLuminanceFromGround(Vector3i position);
+
+    /**
+     * Test whether the object is powered.
+     *
+     * @param position The position of the block
+     * @return Whether powered
+     */
+    boolean isBlockPowered(Vector3i position);
+
+    /**
+     * Test whether the object is indirectly powered.
+     *
+     * @param position The position of the block
+     * @return Whether powered
+     */
+    boolean isBlockIndirectlyPowered(Vector3i position);
+
+    /**
+     * Test whether the face in the given direction is powered.
+     *
+     * @param position The position of the block
+     * @param direction The direction
+     * @return Whether powered
+     */
+    boolean isBlockFacePowered(Vector3i position, Direction direction);
+
+    /**
+     * Test whether the face in the given direction is indirectly powered.
+     *
+     * @param position The position of the block
+     * @param direction The direction
+     * @return Whether powered
+     */
+    boolean isBlockFaceIndirectlyPowered(Vector3i position, Direction direction);
+
+    /**
+     * Get all the faces of this block that are directly powered.
+     *
+     * @param position The position of the block
+     * @return Faces powered
+     */
+    Collection<Direction> getPoweredBlockFaces(Vector3i position);
+
+    /**
+     * Get all faces of this block that are indirectly powered.
+     *
+     * @param position The position of the block
+     * @return Faces indirectly powered
+     */
+    Collection<Direction> getIndirectlyPoweredBlockFaces(Vector3i position);
+
+    /**
+     * Test whether the the block will block the movement of entities.
+     *
+     * @param position The position of the block
+     * @return Blocks movement
+     */
+    boolean isBlockPassable(Vector3i position);
+
+    /**
+     * Test whether the given face of the block can catch fire.
+     *
+     * @param position The position of the block
+     * @param faceDirection The face of the block to check
+     * @return Is flammable
+     */
+    boolean isBlockFlammable(Vector3i position, Direction faceDirection);
```

Comment:
Does this value change based on the doFireTick gamerule? (should clarify in docs)